### ISSUE TITLE
Native runs an `ensure` that was cancelled, when the consuming call is wrapped in `try`

### ISSUE THREAD

**Issue by dritory** [rask-lang/rask#1216]
`mem.linear/L6` says consuming a resource explicitly cancels its registered `ensure`. That works when the consume is a bare statement. Wrap it in `try` and native runs the ensure anyway — a second consume of a value that is already gone.

```rask
struct Oops { public why: string }

extend Oops {
    public func message(self) -> string { return self.why }
}

@resource
struct Conn { public id: i32 }

extend Conn {
    public func finish(take self) -> i32 or Oops {
        println("finished {self.id}")
        return self.id
    }
    public func drop_it(take self) { println("dropped {self.id}") }
}

func run() -> i32 or Oops {
    let c = Conn { id: 1 }
    ensure c.drop_it()
    let v = try c.finish()
    return v
}

func main() {
    let v = run() catch e => {
        println("failed")
        return
    }
    println("{v}")
}
```

```
$ rask run l6.rk
finished 1
dropped 1          <- should not run: `finish` consumed `c`
1

$ rask run --interp l6.rk
finished 1
1
```

The two backends disagree, and the interpreter is right.

## Why

`check_resource_consume` is what emits the `ResourceConsume` that cancels the ensure at cleanup time. It is called from exactly one place:

```rust
// rask-mir/src/lower/stmt.rs
StmtKind::Expr(e) => {
    self.lower_expr(e)?;
    // C1/C2: if this is a consuming method call on an ensure receiver,
    // emit ResourceConsume so the ensure is cancelled at cleanup time.
    self.check_resource_consume(e);
    Ok(())
}
```

A bare `c.close()` statement goes through it. `let v = try c.finish()` doesn't — it's a `Let` whose initializer is a `Try` wrapping the call, and nothing along that path asks. The same is true for `let v = c.finish() catch e => …`, and for a consuming call in a `return`.

It also only looks at the outermost expression, so even as a bare statement `try c.finish()` would miss.

## What it costs

Anything worse than a double `println` depends on what the consuming method does. It bit `spawn`: with

```rask
let b = spawn(|| { return 32i64 })
ensure b.detach()
let y = try b.join()
```

`join` consumes the handle and then `detach` runs on it, and the program segfaults.

That matters because a spawned task handle is a resource (`conc.async/H1`), `try`-joining several of them is the normal shape, and `ensure ….detach()` is exactly what the new E0881 tells you to write for the error path.

## Fix

Call it wherever a statement lowers an expression that could be a consuming call — the `Let`/`Mut` initializer and `return`, not just a bare expression statement — and let it look through `try`, `catch` and `!` before asking whether the call consumes its receiver.

## Found while

Auditing linearity for #882.
